### ISSUE TITLE
Activities not syncing from Garmin Connect

### ISSUE THREAD

**Issue by rammuraly** [cpfair/tapiriik#257]
I had been syncing my run activities from Garmin Connect to Strava and Runkeeper.

From Sep 22, the sync is not happening. Initially it said an error "Some activities could not be synchronised" when I clicked it all activities were shown with tick marks next to services, but the sync has not happened. No specific error message is being given

I have tried reauthorising the services but still no luck, what may be the problem

**Comment by habi** [comment]
Same happens for me.
At first I thought it's because I have a '|' in the description of the activity, but it seems that there's another problem.
My source is Garmin Connect in this case...

![screen shot 2016-09-25 at 19 18 48](https://cloud.githubusercontent.com/assets/1651235/18816959/ea254658-8354-11e6-8d0c-2010849c1be9.PNG)

**Comment by MJunge75** [comment]
The same happens here, no sync from Garmin to Runkeeper and Dropbox.
But when looking at "All activities" I see the Garmin activities in the list with a checkmark that they've synced but they haven't. 

I also reset my Garmin password because a message appeared that for some new security requirements at Garmin the password has to be changed to be able to sync again. But that didn't help.

What's also different to the past is that the 'tapiriik user' isn't listed in the connections anymore.

**Comment by wojtek-b** [comment]
same in my case...

**Comment by pwoidke** [comment]
Same for me. Nothing syncing since Tuesday (activity synced on Tuesday morning, but nothing since then).

**Comment by rob7429** [comment]
Same for me. Was working perfectly and then noticed on Friday, Sept 23rd, that something was broken.

**Comment by bravozulu1** [comment]
Only my cycling has sync'd since July.  My swims and runs do not sync.  Strange.

I also notice taprika is complaining about security requirements for Garmin password complexity.  I rotated the password, just for fun.  But it's still complaining about the login not meeting password requirements.

**Comment by texaschip** [comment]
Same here, not sync-ing Garmin Connect since Sep 15th.  I reset my GC password and re-authorized tapiriik, no luck.  My tapiriik user ID is 54ab5016cbe97e44ee816a41.

**Comment by rob7429** [comment]
To be clear - I have a scenario similar to pwoidke.

Tapiriik seems to be getting the activities from Garmin Connect but is not propagating them to anywhere else. When I view "All Activities", Tapiriik displays the expected activity without any errors. However, the activity doesn't appear in the "Recently Synchronized Activities" list nor do they appear in DropBox or on Strava.

**Comment by rammuraly** [comment]
Yes I can confirm this is exactly my scenario. My activity is read from Garmin Connect but not propogated to Strava or Runkeeper. But shows no error just tick marks showing all done
Any fix coming soon?
Sent from iPhone

		_____________________________

**Comment by wojtek-b** [comment]
Just to note, that reconnecting garmin account does not help in my case (as it is suggested on tapiriik's twitter)

**Comment by meeximum** [comment]
I think the activity-service is down, it returns status 410 Gone :-/

**Comment by meeximum** [comment]
new endpoint seems to be
https://connect.garmin.com/modern/proxy/activity-service/activity/209735936/details

**Comment by RobDickson** [comment]
My swim on 17/09/16 synchronised OK. My cycle on 20/09/16 did not synchronise OK. Nothing has synchronised since then. I have mentioned this by email and Twitter.

Tapiriik must be accessing Garmin Connect OK, because my activities are shown on the Tapiriik website. The problem is that they are all incorrectly marked as private.

My normal process is that Garmin defaults to logging my activities as private and I change those activities that I want to share to public. Tapiriik is set to ignore private activities. When I change the activities to public, I normally rename them too.

Even those that I change immediately after uploading them are still logged as being private and have the original name. Because it still thinks they're private, Tapiriik won't synchronise them.

This might, of course, have been a change by Garmin to its API.

Rob

**Comment by HERSANDI** [comment]
Same since September 22th, Garmin Connect, Dropbox and TP will be possible to resynchronize from that date? Or we'll have to do manually

**Comment by cpfair** [comment]
6cf2a4818c0a02834b1a383982096849cb96e723 seems to fix this issue - it uses a newer API endpoint (which, unfortunately, returns slightly less data - alas).

**Comment by meeximum** [comment]
@cpfair Maybe to use endpoint https://connect.garmin.com/modern/proxy/download-service/export/tcx/activity/{activityId}?

**Comment by RobDickson** [comment]
Is this a suggested fix. or has the fix been applied? Three activities from yesterday (and one from about two months ago that I told you about at the time) haven't synchronised from Garmin to Strava yet. I have manually uploaded my other recent activities, so don't expect them to synchronise.

**Comment by wojtek-b** [comment]
it worked yesterday. Today not anymore. After clicking synchronize it queues for a second, page refreshes and shows "synchronized". But nothing happens.

**Comment by RobDickson** [comment]
My activities were available to be synchronised at about 22:15 yesterday (UK time). I haven't tried a manual synchronise, I was relying on the automatic sync to work.

Thanks for your help trying to fix this.

**Comment by cpfair** [comment]
I discovered two additional issues this morning: 1) I didn't know how `$pullAll` worked (81ed0402aa5555fed60f5e92010b7cea514006e1), so the renew-password error would stick around forever (fixed globally for everyone), and 2) overnight one of the synchronization servers was trying to use an invalid DNS server, so it was "completing" all synchronization jobs (with errors regarding connection failures) instantaneously - also fixed.

**Comment by RobDickson** [comment]
Four activities have just automatically synchronised. The one from two months ago hasn't appeared, but I'm not bothered about that one.

Thanks again.

**Comment by MJunge75** [comment]
Just want to confirm that the sync from/to Garmin works again.
Thanks a lot!

**Comment by rob7429** [comment]
I can also confirm that everything seems to be working as expected now - both a running and cycling activity was successfully synchronised.

Colin - thank you! I'm a software developer and am incredibly impressed with the service and the async architecture behind it. As a community, I hope we can appreciate what Colin and those that have contributed to the project have provided us with.

Cheers,
Rob

**Comment by RobDickson** [comment]
Unfortunately, it seems to be broken for me again.

An OWS on Sunday synchronised OK, but nothing has synchronised since then,

Three private Garmin activities are not shown in Tapiriik's activities (https://tapiriik.com/activities) at all.

One shared Garmin activity is shown as being private in Tapiriik's activities, so is not being synchronised.

I have forced a manual sync, but this hasn't changed anything.

**Comment by jonaslejon** [comment]
Garminc Sync does not work for me either

**Comment by charcarson** [comment]
My Garmin Sync is not working either.
